Javadoc comments and variable name refactoring of:

SKrPl opened the pull request:
1. Classes in org.loklak.client package
2. Classes in org.loklak.tools package

With refference to issue: https://github.com/loklak/loklak_jlib_api/issues/6

SKrPl committed:
Javadoc comments and variable name refactoring of:

1. Classes in org.loklak.client package
2. Classes in org.loklak.tools package

With refference to issue: https://github.com/loklak/loklak_jlib_api/issues/6

djmgit commented on src/org/loklak/client/SearchClient.java:
```diff
@@ -31,39 +31,95 @@
-        return tl;
+        return timeline;
     }
 
```
@SKrPl could please remove this empty line

djmgit commented on src/org/loklak/client/SearchClient.java:
```diff
@@ -31,39 +31,95 @@
-        return tl;
+        return timeline;
     }
 
```
@SKrPl  could you please remove this empty line

SKrPl replied:
After the end of a method a blank line is left (in the middle of source code also), so following that I left a blank line. Same goes for other blank lines left after method completion.

djmgit commented on src/org/loklak/client/SuggestClient.java:
```diff
@@ -31,51 +31,103 @@
-        return rl;
+        return resultList;
     }
 
```
here too

djmgit commented on src/org/loklak/tools/JsonCompatHelper.java:
```diff
@@ -1,51 +1,58 @@
+        }
     }
-  }
 
```
this too and the other empty lines between braces

sudheesh001 commented on src/org/loklak/tools/JsonCompatHelper.java:
```diff
@@ -1,51 +1,58 @@
- *  If not, see <http://www.gnu.org/licenses/>.
+ * JsonCompatHelper
+ * Copyright 06.06.2016 by Marc Nause, @nause_marc
+ * <p>
```
Is there a reason to add this `<p>` in the copyright header? These are standardized and should be left as is.

SKrPl replied:
Sorry this happened by mistake. JsonCompatHelper class has 2 spaces as indentation, while fixing that IDE automatically changed it. I will remove the `<p>` tags from the header.

sudheesh001 commented on src/org/loklak/tools/JsonCompatHelper.java:
```diff
@@ -1,51 +1,58 @@
 import org.json.JSONArray;
 import org.json.JSONObject;
 
+import java.util.Collection;
```
Is there a reason for updating this import and moving it to a different line?

SKrPl replied:
Happened due to *Automatic Reformat* of IDE (the same reason for the previous comment regarding addition of `<p>` tags) . Should I change it back ?

vibhcool replied:
@sudheesh001 Shouldn't the java + 3rd party imports and project imports be kept separated?

sudheesh001 replied:
Ideally they should be, but my question here was why was the line deleted and added elsewhere. The culprit is the IDE. that answers it. It's alright to leave this.

sudheesh001 commented on src/org/loklak/tools/JsonCompatHelper.java:
```diff
@@ -1,51 +1,58 @@
+                            new Class[]{String.class, Collection.class}) != null;
+        } catch (NoSuchMethodException | SecurityException e) {
+            hasCollectionMethod = false;
+        }
```
structuring the code like this makes it look literally unreadable. Can you move the `hasCollectionMethod` to a single line like before while updating the string to the `JSON_OBJECT_METHOD_NAME`?

SKrPl replied:
Will the following one look better (considering 100 chars) ?
http://pasteboard.co/2GcPRUW7.png

sudheesh001 replied:
I think this looks much better.

sudheesh001 commented on src/org/loklak/tools/JsonCompatHelper.java:
```diff
@@ -1,51 +1,58 @@
+    public static JSONObject put(
+            final JSONObject jsonObject,
+            final String key,
+            final Collection<?> value) {
```
These are just 3 parameters, It might as well be in a single line instead of breaking it down this way. It's a good practice to break them down only if they overshoot a decent line limit of 80 in most cases but in those cases (with lots of parameters) it might as well be okay to put each parameter on a new line.

SKrPl replied:
Can I do something like ...

`public static JSONObject put(`

            `final JSONObject jsonObject, final String key, final Collection<?> value) {`

This way it won't exceed 100 chars.

sudheesh001 replied:
That's alright.

sudheesh001 commented on src/org/loklak/tools/JsonIO.java:
```diff
@@ -26,20 +26,43 @@
+    public static JSONObject pushJson(
+            String requestURL,
+            String jsonObjectKey,
+            JSONObject jsonObject) throws IOException, JSONException {
```
Same comment here. Put them in one line.

SKrPl replied:
Can I put the exceptions in 2nd line ?

sudheesh001 replied:
Sounds good.

sudheesh001 commented on src/org/loklak/tools/NetworkIO.java:
```diff
@@ -35,77 +35,110 @@
+     * @throws IOException
+     * @throws JSONException
+     */
+    public static StringBuilder pushString(String requestURL, String jsonDataKey, String bodyText)
```
Looking at the rest of your code where you put them into multiple lines, this becomes inconsistent. It will be better if all these functions look exactly like this, one liners with exceptions thrown on next line or even better if on the same line.

SKrPl replied:
I want to restrict number of characters per line to 100, that is why I am breaking it down to multiple lines. So, that later when I add TravisCI I can add a check for 100 chars/line. What do you suggest on this ?

I have provided alternate options (where a method has less parameters but it consumes more than 100 chars). Please give your suggestions on those. Or  is it okay to be lenient on this ?

sudheesh001 replied:
I've left some feedback on that, they look good with the changes proposed.

sudheesh001 commented on src/org/loklak/tools/NetworkIO.java:
```diff
@@ -35,77 +35,110 @@
-        PrintWriter writer = new PrintWriter(new OutputStreamWriter(outputStream, charset), true);
+        URLConnection urlConnection = (url).openConnection();
+        HttpURLConnection connection =
+                requestURL.startsWith("https") ? (HttpsURLConnection) urlConnection :
```
start immediately after the `=` not on a new line.

sudheesh001 commented on src/org/loklak/client/SearchClient.java:
```diff
@@ -31,39 +31,95 @@
+                                  final String source,
+                                  final int count,
+                                  final int timezoneOffset,
+                                  final long timeout) throws IOException {
```
This looks good. really long parameter methods can be broken down to have params on new lines. 👍

sudheesh001 reviewed (changes requested):
Please find the review comments.

SKrPl commented on src/org/loklak/client/PushClient.java:
```diff
@@ -28,20 +28,32 @@
-        JSONObject json = JsonIO.pushJson(hoststub + "/api/push.json", "data", timeline.toJSON(false));
-        if (json != null) return json;
-        return null;
+        if (hostServerUrl.endsWith("/"))
```
TODO: Keep the condition of `if` clause inside curly brace.
TODO: Rectify this for other cases.

vibhcool commented on src/org/loklak/client/PushClient.java:
```diff
@@ -28,20 +28,32 @@
-        if (json != null) return json;
-        return null;
+        if (hostServerUrl.endsWith("/"))
+            hostServerUrl = hostServerUrl.substring(0, hostServerUrl.length() - 1);
```
here this must be enclosed in braces of if condition

SKrPl replied:
Yes, I found this just when I shared link with you. Will do the changes :+1:

vibhcool commented on src/org/loklak/client/SearchClient.java:
```diff
@@ -31,39 +31,95 @@
+
+        JSONObject json = JsonIO.loadJson(searchApiUrl);
+
+        if (json.length() == 0)
```
here too, please also see in other files too

SKrPl committed:
Removes <p> tags and Changes in method definition

1. Removes <p> tags from the header of JsonCompatHelper.java
2. Params are shifted to 2nd line of method definition for methods that have
less parameters but use more than 100 chars.
3. Exceptions thrown are shifted to 2nd line of method definition if it
exceeds 100 chars.
4. Single line body of "if" statements are enclosed inside curly braces.

SKrPl committed:
Adds unit-tests for classes in client package

Private fields are accessed using Reflection so that they can be used in
unit-testing.

Solves adding of unit-test for client package, as mentioned in
issue: https://github.com/loklak/loklak_jlib_api/issues/6

SKrPl committed:
Adds unit-tests for classes in tools package

Solves a part of: https://github.com/loklak/loklak_jlib_api/issues/6

1. TimelineStub is created, so that it can be used to test POST request
methods in JsonIO and NetworkIO (uses push API of loklak).
2. Constants class contains values which are used for testing (mostly
contains the JSONObject key of API replies).
3. Small changes in classes of test.org.loklak.client package (using the
created TimelineStub and Constant classes).

SKrPl commented:
Have added unit test for client and tools package, please review it.

vibhcool commented on test/org/loklak/client/SuggestClientTest.java:
```diff
@@ -0,0 +1,132 @@
+        JSONArray jsonQueryArray = jsonObject.getJSONArray(JSON_KEY_QUERIES);
+        if (jsonQueryArray.length() > 0) {
+            JSONObject queryObject = jsonQueryArray.getJSONObject(0);
+            assertTrue(queryObject.has(QUERY_KEY_QUERY));
```
try using assertThat for better readability
see this -> https://objectpartners.com/2013/09/18/the-benefits-of-using-assertthat-over-other-assert-methods-in-unit-tests/

vibhcool commented:
@SKrPl may be Tests should be in different PR

SKrPl committed:
Uses AssertThat replacing AssertTrue

Creates a custom Hamcrest matcher "hasKey" in Utility class so that it
can be used with "assertThat" method.

SKrPl commented:
@vibhcool As I refactored a lot of variables, so I was suggested to write unit-tests for them. Thus code refactoring and unit-test in same PR.
I have done the requested changes :smile: .

vibhcool commented:
@SKrPl i got errors https://dpaste.de/ZGBk on running command mvn install, please see

SKrPl commented:
@vibhcool did the required changes and rebased it.

vibhcool commented:
There is no error or failure in tests, but on running tests IOException appears due to 403 response error -> https://dpaste.de/H4Q9

Also i am getting error -> https://dpaste.de/RcGN on running command `mvn build`

vibhcool reviewed (changes requested):
There is no error or failure in tests, but on running tests IOException appears due to 403 response error -> https://dpaste.de/H4Q9

Also i am getting error -> https://dpaste.de/RcGN on running command `mvn build`
please see

SKrPl commented:
@vibhcool `search` method of `TwitterScraper.java` is used in `TimelineStub.java` which handles the exception by printing the stack trace. There is no failure of test because if the request is not complete we won't get any JSONObject, so we can't say that the test fails.

And regarding `mvn build`, to build `mvn compile` is used, there is no such goal called `build`. In case a '.jar' file is required to generate `mvn package` is used.

vibhcool reviewed (approved):
IMO, since we know the reason about why Exception happened and there is no error, could you just output line with reason why failure occurred, instead of stacktrace. Stacktrace is useful when there is error and is needed to be resolved. You may improve this if you want.

My bad, I got confused about gradle and mvn here, will take care next time. 😅 

Rest looks fine 👍
